### PR TITLE
Amqp provider based on #3812

### PR DESCRIPTION
**Pull request by ngtive**
### What change does this PR introduce?

This PR will add amqp protocol provider

### Why was this change needed?
Based on
#3812

### CODE REVIEW

File: libs/shared/src/consts/providers/channels/sms.ts [hunk 1]
```diff
@@ -1,23 +1,24 @@
 import { IProviderConfig } from '../provider.interface';
 import {
-  gupshupConfig,
```
Comment (yiiman-dev) [inline libs/shared/src/consts/providers/channels/sms.ts]: Please revert sorting

File: libs/shared/src/consts/providers/credentials/provider-credentials.ts [hunk 2]
```diff
@@ -740,3 +740,36 @@ export const sendchampConfig: IConfigCredentials[] = [
   },
   ...smsConfigBase,
 ];
+
+export const amqpSmsConfig: IConfigCredentials[] = [
+  {
+    key: CredentialsKeyEnum.ApiKey,
```
Comment (yiiman-dev) [inline libs/shared/src/consts/providers/credentials/provider-credentials.ts]: Please remove this key
you can use fixed string ( "novu-sms") as queue  name

**Comment by unicodeveloper** [comment]
@p-fernandez Please can you help check this out?

**Comment by scopsy** [comment]
Closing due to inactivity

**Pull request closed without merging**